lib/tools.js

```diff
@@ -2410,17 +2410,25 @@ function parseGithubPathname(pathname) {
 
 /**
  * Removes properties which are given by preserve
- * @param {object} preserve - object which has true entries for all attributes which should be removed from currObj
+ * @param {object} preserve - object which has true entries (or array of selected attributes) for all attributes which should be removed from currObj
  * @param {object} oldObj - old object
  * @param {object} newObj - new object
  */
 function removePreservedProperties(preserve, oldObj, newObj) {
     for (const prop of Object.keys(preserve)) {
         if (isObject(preserve[prop]) && newObj[prop]) {
```

Comment:
what if `newObj[prop]` is **not** an object?

Reply:
Guess it makes sense to check it too